pkg/controllers/webhookauthenticator/webhookauthenticator_controller.go

```diff
@@ -0,0 +1,176 @@
+	"github.com/openshift/library-go/pkg/operator/v1helpers"
+
+	"github.com/openshift/cluster-authentication-operator/pkg/operator/assets"
+)
+
+// webhookAuthenticatorController makes sure that the webhook token authenticators
+// are configured so that they point to the oauth-apiserver's tokenvalidation
+// endpoint. That is, if the type of authentication in the auth config is set t
+// be the integrated authentication
+type webhookAuthenticatorController struct {
+	authentication  configv1client.AuthenticationInterface
+	serviceAccounts corev1client.ServiceAccountsGetter
+
+	saLister  corev1listers.ServiceAccountLister
+	svcLister corev1listers.ServiceLister
+	secrets   corev1client.SecretsGetter
+
+	operatorConfigClient operatorconfigclient.AuthenticationsGetter
+	operatorClient       v1helpers.OperatorClient
+}
+
+func NewWebhookAuthenticatorController(
+	kubeInformersForTargetNamespace informers.SharedInformerFactory,
+	configInformer configinformers.SharedInformerFactory,
+	secrets corev1client.SecretsGetter,
+	serviceAccounts corev1client.ServiceAccountsGetter,
+	authentication configv1client.AuthenticationInterface,
+	operatorConfigClient operatorconfigclient.AuthenticationsGetter,
+	operatorClient v1helpers.OperatorClient,
+	recorder events.Recorder,
+) factory.Controller {
+	c := &webhookAuthenticatorController{
+		secrets:              secrets,
+		serviceAccounts:      serviceAccounts,
+		svcLister:            kubeInformersForTargetNamespace.Core().V1().Services().Lister(),
+		saLister:             kubeInformersForTargetNamespace.Core().V1().ServiceAccounts().Lister(),
+		authentication:       authentication,
+		operatorConfigClient: operatorConfigClient,
+		operatorClient:       operatorClient,
+	}
+	return factory.New().WithInformers(
+		kubeInformersForTargetNamespace.Core().V1().ServiceAccounts().Informer(),
+		kubeInformersForTargetNamespace.Core().V1().Services().Informer(),
+		kubeInformersForTargetNamespace.Core().V1().Secrets().Informer(),
+		configInformer.Config().V1().Authentications().Informer(),
+	).ResyncEvery(30*time.Second).
+		WithSync(c.sync).
+		WithSyncDegradedOnError(operatorClient).
+		ToController("WebhookAuthenticatorController", recorder.WithComponentSuffix("webhook-authenticator-controller"))
+}
+
+func (c *webhookAuthenticatorController) sync(ctx context.Context, syncCtx factory.SyncContext) error {
+	authConfig, err := c.authentication.Get(ctx, "cluster", metav1.GetOptions{})
+	if err != nil {
+		return err
+	}
+
+	if authConfig.Spec.Type != configv1.AuthenticationTypeIntegratedOAuth && len(authConfig.Spec.Type) != 0 {
+		// not using integrated openshift auth
+		return nil
+	}
+
+	oauthAPIsvc, err := c.svcLister.Services("openshift-oauth-apiserver").Get("api")
+	if err != nil {
+		return fmt.Errorf("failed to retrieve service openshift-oauth-apiserver/oauth-apiserver: %w", err)
+	}
+
+	kubeConfigSecret, err := c.ensureKubeConfigSecret(ctx, oauthAPIsvc, syncCtx.Recorder())
```

Comment:
if we eventually do this with a CSR, this part will be done in a separate config loop

Reply:
> if we eventually do this with a CSR, this part will be done in a separate config loop

when this happens, you'll want to have an existence check before setting this value so we don't create a situation where a bad path stops the kube-apiserver from starting.

Reply:
I expect to have a separate controller that creates a CSR, stores its private key in a secret, waits for the CSR to get approved (different controller), and then it synces the cert from the CSR into the secret with the key.